File: reverse.whois/header-auth/ruby/reverse_whois_api_v2_header_auth.rb
require 'json'
require 'net/https'
require 'openssl'
require 'uri'
require 'yaml' # only needed to print the returned result in a very pretty way

url = 'https://reverse-whois-api.whoisxmlapi.com/api/v2'

########################
# Fill in your details #
########################
key = 'Your reverse whois api key'

params_advanced = {
  advancedSearchTerms: [
    {
      field: 'RegistrantContact.Name',
      term: 'Test'
    }
  ],
  mode: 'purchase',
  sinceDate: '2018-07-15'
}

params_basic = {
  basicSearchTerms: {
    include: %w[
      test
    ],
    exclude: %w[
      whois
      api
    ]
  },
  mode: 'purchase',
  sinceDate: '2018-07-15'
}

uri = URI.parse(url)
http = Net::HTTP.new(uri.host, uri.port)

# Connect using ssl
http.use_ssl = true
http.verify_mode = OpenSSL::SSL::VERIFY_NONE
request = Net::HTTP::Post.new(uri.request_uri)

# Set headers
request.add_field('Content-Type', 'application/json')
request.add_field('Accept', 'application/json')
request.add_field('X-Authentication-Token', key)

# Basic search

request.body = params_basic.to_json
response = http.request(request)

# Print pretty parsed json
puts 'Basic:'
puts JSON.parse(response.body).to_yaml

# Advanced search

request.body = params_advanced.to_json
response = http.request(request)

puts "\nAdvanced:"
puts JSON.parse(response.body).to_yaml

File: reverse.whois/header-auth/node/reverse_whois_api_v2_header_auth.js
var https = require('https');

// Fill in your details
var api_key = 'Your reverse whois api key';

// Build the post string

var post_data_advanced = {
    advancedSearchTerms: [
        {
            field: 'RegistrantContact.Name',
            term: 'Test'
        }
    ],
    mode: 'purchase',
    sinceDate: '2018-07-12'
};

var post_data_basic = {
    basicSearchTerms: {
        include: [
            'test',
            'US'
        ],
        exclude: [
            'Europe',
            'EU'
        ],
    },
    mode: 'purchase',
    sinceDate: '2018-07-12'
};

function api_call(data, callback)
{

    var body = '';

    var opts = {
        hostname: 'reverse-whois-api.whoisxmlapi.com',
        path: '/api/v2',
        method:'POST',
        headers: {
            'Content-Type': 'application/json',
            'Accept': 'application/json',
            'Content-Length': JSON.stringify(data).length,
            'X-Authentication-Token': api_key,
        }
    };

    var req = https.request(opts, function(response) {
        response.on('data', function(chunk) {
            body += chunk;
        });
        response.on('end', function() {
            callback(JSON.parse(body));
        });
    });

    req.on('error', function(e) {
        console.error(e);
        process.exit(1);
    });

    req.write(JSON.stringify(data));
    req.end();
}

// Send requests and log responses

api_call(post_data_basic, function(body) {
    console.log('Basic:');
    console.log(body);

    api_call(post_data_advanced, function(body) {
        console.log('Advanced:');
        console.log(body);
    });
});